### PR TITLE
drivers:platform:pico:pico_spi: Add spi_transfer API

### PR DESCRIPTION
**Pull request by rbolboac**
Added spi_transfer API for pico platform which allows to send multiple spi messages with chip select either asserted or de-asserted.

### CODE REVIEW

**Review by buha — APPROVED** [review]

**Comment by rbolboac** [comment]
- removed blank line

File: drivers/platform/pico/pico_spi.c [hunk 1]
```diff
@@ -241,11 +250,65 @@ int32_t pico_spi_write_and_read(struct no_os_spi_desc *desc,
...
+			gpio_pull_down(pico_spi->spi_cs_pin);
+		else
+			gpio_pull_up(pico_spi->spi_cs_pin);
+
+		spi_write_read_blocking(pico_spi->spi_instance, msgs[i].tx_buff,
+					msgs[i].rx_buff, msgs[i].bytes_number);
+
+		if (msgs[i].cs_change) {
+			gpio_set_function(pico_spi->spi_cs_pin, GPIO_FUNC_SPI);
+		}
```
Comment (CiprianRegus) [inline drivers/platform/pico/pico_spi.c:298]: Brackets are not needed.

**Review by CiprianRegus — APPROVED** [review]
Looks good, just styling.

**Comment by rbolboac** [comment]
- removed unnecessary brackets

**Review by amiclaus — APPROVED** [review]

**Pull request merged**